practice/validator/site.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+/**
+ * The Validator takes in the user's input as they type into any given text box
+ * in the form and uses kestroke events to visually que the user that their
+ * input is either valid or not, by filling the field with a yellow background
+ * while invalid.
+ */
+
+/**
+ * Adds the invalid class to the given field (distinguished by a class).
+ *
+ * The invalid class has a yellow background in the site.css file.
+ */
+function makeFieldYellow(inputclass) {
+  $('input.' + inputclass).addClass('invalid');
+}
+
+/**
+ * Removes the invalid class from a given field (distinguished by a class).
+ */
+function removeYellowFromField(inputclass) {
+  $('input.' + inputclass).removeClass('invalid');
+}
+
+/**
+ * Takes in a field (distinguished by a class) and the entry that has been
+ * typed into the textbox of the given field, and returns a boolen representing
+ * whether or not the entry is valid for the given field.
+ */
+function checkIfValid(inputclass, entry) {
+  if (inputclass === 'fullname') {
+    return /[a-z]+\s[a-z]+/i.test(entry);
+  } else if (inputclass === 'birthday') {
+    return /\d{4}\-(\d{2})\-(\d{2})/.test(entry);
+  } else if (inputclass === 'phone-number') {
+    return /(\d{3})\-(\d{3})\-(\d{4})/.test(entry);
+  }
+}
```

Comment:
This is really three separate functions with three separate names that are being overloaded into whatever the content of `inputClass` is. If these were broken apart into three functions, then the linter can check when you misspell them or you don't have strong coupling between your class names and the ability to validate text.